File: hazelcast-continuous-query/src/hazelcast_continuous_query/book.clj
(ns hazelcast-continuous-query.book
  (:import (java.io Serializable)
           (java.util Map)))

(gen-class :name hazelcast-continuous-query.book.Book
           :implements [java.io.Serializable]
           :state state
           :init init
           :constructors {[java.util.Map] []}
           :methods [[getIsbn13 [] String]
                     [getName [] String]
                     [getPrice [] int]
                     [getPublishDate [] String]
                     [getCategory [] String]
                     [isOutOfPrint [] boolean]])

(defn -init [underlying]
  [[] underlying])

(defn -getIsbn13 [this]
  (:isbn13 (. this state)))

(defn -getName [this]
  (:name (. this state)))

(defn -getPrice [this]
  (:price (. this state)))

(defn -getPublishDate [this]
  (:publish-date (. this state)))

(defn -getCategory [this]
  (:category (. this state)))

(defn -isOutOfPrint [this]
  (:out-of-print (. this state)))

(defn -toString [this]
  (str (. this state)))

File: hazelcast-entry-processor-re/src/test/java/org/littlewings/hazelcast/entryprocessor/Book.java
package org.littlewings.hazelcast.entryprocessor;

import java.io.Serializable;

public class Book implements Serializable {
    private static final long serialVersionUID = 1L;

    String isbn;
    String title;
    int price;
    Category category;

    public static Book create(String isbn, String title, int price, Category category) {
        return new Book(isbn, title, price, category);
    }

    public Book(String isbn, String title, int price, Category category) {
        this.isbn = isbn;
        this.title = title;
        this.price = price;
        this.category = category;
    }

    public String getIsbn() {
        return isbn;
    }

    public String getTitle() {
        return title;
    }

    public int getPrice() {
        return price;
    }

    public Category getCategory() {
        return category;
    }
}
